presto-main/src/test/java/io/prestosql/tests/ReportUnannotatedMethods.java

```diff
@@ -14,27 +14,44 @@
 package io.prestosql.tests;
 
 import com.google.common.reflect.TypeToken;
-import org.testng.IClassListener;
-import org.testng.ITestClass;
+import io.airlift.log.Logger;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+
+import javax.annotation.concurrent.ThreadSafe;
 
 import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Set;
 import java.util.stream.Stream;
 
 import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.Sets.newConcurrentHashSet;
 import static java.lang.String.format;
 import static java.util.stream.Collectors.joining;
 
+@ThreadSafe
 public class ReportUnannotatedMethods
-        implements IClassListener
```

Comment:
I would stick with `IClassListener` and use `System.exit` if raising exception from listener does not fail the build